azerothcore/azerothcore-wotlk
# fix(CMake): Deprecation warning

## CHANGES PROPOSED:
-  Fix deprecation warning
-  Properly set minimum version (see https://cliutils.gitlab.io/modern-cmake/chapters/basics.html#minimum-version)


## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1261



## TESTS PERFORMED:

Manual tested locally:
- macOS Catalina: `CMake version: 3.17.3`
- macOS Catalina: `CMake version: 3.18.4`

Tested automatically in CI:
- Ubuntu 20.04: `CMake version: 3.17.0`
- Ubuntu 18.04: `CMake version: 3.17.0`
- Docker: `CMake version: 3.16.3`
- Windows: `CMake version: 3.18.4`
- macOS Catalina: `CMake version: 3.18.4`



## HOW TO TEST THE CHANGES:

- Run CMake again
- Rebuild
- Make sure the build was successful

## KNOWN ISSUES AND TODO LIST:


* I think it's enough to test Linux in the CI as the process is the same as you would normally do locally, also we run the dry run on Linux. While on Windows I'd still prefer if someone else tests it manually. 


## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
